crates/compiler/src/lib.rs

```diff
@@ -237,6 +245,18 @@ mod v2 {
 
         Ok(casm_definition)
     }
+
+    pub(super) fn casm_class_hash(casm_definition: &[u8]) -> anyhow::Result<CasmHash> {
```

Comment:
Separate question: this is presumably used to verify p2p casm data received matches the expected casm hash. Do we know if this works for older casm classes?

Reply:
It should. If it's not, execution is broken for older classes as well since blockifier uses the exact same type from the compiler to deserialize CASM class definitions.

Reply:
I'm sure it can deserialize them; but that doesn't mean it can calculate its hash